front/app/api/internal_comments/useInternalComments.ts

```diff
@@ -0,0 +1,67 @@
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import { getPageNumberFromUrl } from 'utils/paginationUtils';
+import internalCommentKeys from './keys';
+import {
+  InternalCommentsKeys,
+  IInternalCommentParameters,
+  IInternalComments,
+  IInternalCommentQueryParameters,
+} from './types';
+
+const getInternalCommentsEndpoint = ({
+  ideaId,
+  initiativeId,
+  userId,
+  commentId,
+}: IInternalCommentParameters) => {
```

Comment:
This works well but I think it might be brittle. From what I can understand, the ideal scenario should have only one of these (ideaId, initiativeId, userId, commentId) filled. This means that in an illogical state, if you provide an ideaId, then you always get the idea endpoint even if you provide the commentId. We could tweak this by using discriminated unions. See [here](https://basarat.gitbook.io/typescript/type-system/discriminated-unions). It works for now so it is not a blocker but it would be a hard bug to track if we leave it as is

Something like 
```
export type IInternalCommentParameters =
  | { type: 'idea'; ideaId: string }
  | { type: 'initiative'; initiativeId: string }
  | { type: 'user'; userId: string }
  | { type: 'comment'; commentId: string };

const getInternalCommentsEndpoint = (params: IInternalCommentParameters) => {
  const { type, ideaId, initiativeId, userId, commentId } = params;

  switch (type) {
    case 'idea':
      return `ideas/${ideaId}/internal_comments`;
    case 'initiative':
      return `initiatives/${initiativeId}/internal_comments`;
    case 'comment':
      return `internal_comments/${commentId}/children`;
    case 'user':
    default:
      return `users/${userId}/internal_comments`;
  }
};
```